Src/ILGPU.Algorithms.Tests/Generic/Xunit/Sdk/SkippableFactDiscoverer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
```

Comment:
Are these official extensions from the Xunit.SDK?

Reply:
They are from the official samples.

There is a link in the commit message:
https://github.com/xunit/samples.xunit/tree/5334ee9cf4a81f40dcb4cafabfeb098a555efb3d/DynamicSkipExample

I made some simple modifications because the sample is slightly out-of-date.

In future, I was hoping to modify the code to add ILGPU-specific exception handling (e.g. to automatically skip a test if the intrinsic is not supported)

Should I move it into a separate project?

Reply:
I am always worried about integrating third-party source code into our repositories because of licenses and copyright issues that might pop-up in the future. Can we live without this code :)?

Reply:
Done - test cases have been modified, and history rewritten to remove the third-party source code.